files/scripts/syncd.sh

```diff
@@ -92,18 +92,9 @@ function waitplatform() {
 
     BOOT_TYPE=`getBootType`
     if [[ x"$sonic_asic_platform" == x"mellanox" ]]; then
-        if [[ x"$BOOT_TYPE" = @(x"fast"|x"warm"|x"fastfast") ]]; then
-            PMON_TIMER_STATUS=$(systemctl is-active pmon.timer)
-            if [[ x"$PMON_TIMER_STATUS" = x"inactive" ]]; then
-                systemctl start pmon.timer
-            else
-                debug "PMON service is delayed by a timer for better fast/warm boot performance"
-            fi
-        else
-            debug "Starting pmon service..."
-            /bin/systemctl start pmon
-            debug "Started pmon service"
-        fi
+        debug "Starting pmon service..."
```

Comment:
@vadymhlushko-mlnx Why will Systemd not start the PMON early? Why not remove "After=Syncd" for your platform? https://github.com/sonic-net/sonic-buildimage/blob/6d0cd5519560f48cb2a7421588d100c617996eb7/files/build_templates/pmon.service.j2#L6

Reply:
@vadymhlushko-mlnx  would you please check the comments?

Reply:
@vadymhlushko-mlnx There is already a dependency between Xcvrd and Swss. Xcvrd waits for [PortInitDone](https://github.com/sonic-net/sonic-platform-daemons/blob/master/sonic-xcvrd/xcvrd/xcvrd.py#L2423) and untill portsyncd sets its after all host interfaces are created: https://github.com/sonic-net/sonic-swss/blob/19410232523283d99eaa211979cc1e58b6cb262f/portsyncd/portsyncd.cpp#L134

We are holding complete PMON init just for Xcvrd which doesn't seems right. Can we use PortInitDone as a synchronization?

Reply:
@prgeor kind reminder, I'm waiting for your comment or approve

Reply:
@vadymhlushko-mlnx Please explain more about:-
"@prgeor XCVRD should be started after we create_switch and initialize all the ports, that is why we have a dependency for the syncd."

Reply:
@prgeor if we remove the `After=syncd.service` from the `pmon.service` file - the `PMON` is going to start parallel to `SYNCD`. It could lead to such a scenario - if in the `PMON,` someone is going to do some ASIC-related operation while in parallel the `SYNCD` is doing `create_switch` and resetting the ASIC - such flow will introduce a degradation.

Reply:
@vadymhlushko-mlnx synchronization via systemd service file is fine. Why do we need to start pmon from synch.sh?

Reply:
@vadymhlushko-mlnx let systemd do the synchronization between syncd and pmon service

Reply:
@prgeor Some services are started by shell scripts, some via systemd. Both approaches work in SONiC. Starting from shell script gives more flexibility

Reply:
@prgeor I've removed unused code from the `syncd.sh` as you requested

Reply:
@vadymhlushko-mlnx OK. is the PR still in draft or ready for review?